service/history/queue/timer_queue_processor.go

```diff
@@ -225,23 +239,37 @@ func (t *timerQueueProcessor) FailoverDomain(domainIDs map[string]struct{}) {
 		}
 	}
 
+	if standbyClusterName != t.currentClusterName {
+		t.logger.Debugf("Timer queue failover will use minLevel: %v from standbyClusterName: %s", minLevel, standbyClusterName)
+	} else {
+		t.logger.Debugf("Timer queue failover will use minLevel: %v from current cluster: %s", minLevel, t.currentClusterName)
+	}
+
 	maxReadLevel := time.Time{}
 	actionResult, err := t.HandleAction(context.Background(), t.currentClusterName, NewGetStateAction())
 	if err != nil {
-		t.logger.Error("Timer Failover Failed", tag.WorkflowDomainIDs(domainIDs), tag.Error(err))
+		t.logger.Error("Timer failover failed while getting queue states", tag.WorkflowDomainIDs(domainIDs), tag.Error(err))
 		if err == errProcessorShutdown {
 			// processor/shard already shutdown, we don't need to create failover queue processor
 			return
 		}
 		// other errors should never be returned for GetStateAction
 		panic(fmt.Sprintf("unknown error for GetStateAction: %v", err))
 	}
+
+	var maxReadLevelQueueLevel int
 	for _, queueState := range actionResult.GetStateActionResult.States {
 		queueReadLevel := queueState.ReadLevel().(timerTaskKey).visibilityTimestamp
 		if maxReadLevel.Before(queueReadLevel) {
 			maxReadLevel = queueReadLevel
+			maxReadLevelQueueLevel = queueState.Level()
```

Comment:
this seems reasonable, but also is it part of this conceptual change? I wonder if it should be a separate PR?

Reply:
This is a side change to improve debugging of this component. I think it's small enough change to bundle here.